### DISCUSSION TITLE
What permissions do I need when creating an ECS agent using the Cloudformation stack?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18053] [Q&A]
Hi all, I'm trying to create the <https://docs.dagster.io/dagster-cloud/deployment/agents/amazon-ecs/creating-ecs-agent-new-vpc#prerequisites|ECS agent> using the CF stack, and I'm running into permissions issues trying to set up the ECS role. I tried using `arn:aws:iam::&lt;account&gt;:role/aws-service-role/ecs.amazonaws.com/AWSServiceRoleForECS`  and got `AWSServiceRoleForECS is invalid or cannot be assumed`, so maybe I need a role that CF has permissions to assume as the principal?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C014UDS8LAV/p1700078914195909?thread_ts=1700078914.195909&cid=C014UDS8LAV).*

**Comment by prha** [comment]
Not tested, but it looks like there might be a few workarounds you might be able to try here: https://stackoverflow.com/a/68384249?  Maybe going to the ECS console manually after the initial deployment attempt might work?
